lib/puppet/network/http/connection.rb

```diff
@@ -189,6 +189,32 @@ def request_with_redirects(request, options)
             request.each do |header, value|
               current_request[header] = value
             end
+          when 503, 429
+            retry_after = current_response['Retry-After']
+
+            if retry_after.nil?
+              # Set return value and advance to next loop.
+              response = current_response
+              next
+            end
+
+            begin
+              retry_sleep = Integer(retry_after)
```

Comment:
Technically, we should also accept a HTTP Date as a valid time to retry the request:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

However, the Puppet Server implementation of sending a 503 when overloaded uses an integer number of seconds so the marginal utility gained by also accepting dates would be very small at this time.